server/database.go

```diff
@@ -92,19 +93,33 @@ func getAllApplicationDeployments(db *sql.DB, a *models.Application) ([]*models.
 }
 
 func getApplicationDeployments(db *sql.DB, a *models.Application, limit int) ([]*models.Deployment, error) {
-	deployments := []*models.Deployment{}
-
 	rows, err := db.Query(applicationDeploymentsStmt, a.Name, limit)
 	if err != nil {
-		return deployments, err
+		return nil, err
+	}
+	defer rows.Close()
+
+	return readApplicationDeployments(rows)
+}
+
+func getApplicationDeploymentsByTarget(db *sql.DB, a *models.Application, target string) ([]*models.Deployment, error) {
```

Comment:
How about using a `Target` instead of a `string` for the target here? We have the type and this way we could let the type system work for us.